Objective-C/Tests/VectorSearchTest.m

```diff
@@ -0,0 +1,142 @@
+    AssertEqual(config.metric, kCBLDistanceMetricCosine);
+    AssertEqual(config.minTrainingSize, 100);
+    AssertEqual(config.maxTrainingSize, 200);
+}
+
+- (void) _testDimensionsValidation {
+    CBLVectorIndexConfiguration* config1 = [[CBLVectorIndexConfiguration alloc] initWithExpression:@"vector" dimensions: 1 centroids: 10];
+    AssertNotNil(config1);
+    
+    CBLVectorIndexConfiguration* config2 = [[CBLVectorIndexConfiguration alloc] initWithExpression:@"vector" dimensions: 2048 centroids: 10];
+    AssertNotNil(config2);
+    
+    [self expectException: NSInvalidArgumentException in:^{
+        (void) [[CBLVectorIndexConfiguration alloc] initWithExpression:@"vector" dimensions: 0 centroids: 10];
+        NSLog(@"Invalid Argument test for dimensions = 0");
+    }];
+    
+    [self expectException: NSInvalidArgumentException in:^{
+        (void) [[CBLVectorIndexConfiguration alloc] initWithExpression:@"vector" dimensions: 2049 centroids: 10];
+        NSLog(@"Invalid Argument test for dimensions = 2049");
+    }];
+}
+
+- (void) _testCentroidsValidation {
+    CBLVectorIndexConfiguration* config1 = [[CBLVectorIndexConfiguration alloc] initWithExpression:@"vector" dimensions: 300 centroids: 1];
+    AssertNotNil(config1);
+    
+    CBLVectorIndexConfiguration* config2 = [[CBLVectorIndexConfiguration alloc] initWithExpression:@"vector" dimensions: 300 centroids: 64000];
+    AssertNotNil(config2);
+    
+    [self expectException: NSInvalidArgumentException in:^{
+        (void) [[CBLVectorIndexConfiguration alloc] initWithExpression: @"vector" dimensions: 300 centroids: 0];
+        NSLog(@"Invalid Argument test for centroids = 0");
+    }];
+    
+    [self expectException: NSInvalidArgumentException in:^{
+        (void) [[CBLVectorIndexConfiguration alloc] initWithExpression: @"vector" dimensions: 300 centroids: 64001];
+        NSLog(@"Invalid Argument test for centroids = 64001");
+    }];
+}
+
+- (void) _testCreateVectorIndex {
+    NSError* error;
+    CBLCollection* collection = [_db collectionWithName: @"words" scope: nil error: nil];
+    CBLVectorIndexConfiguration* config = [[CBLVectorIndexConfiguration alloc] initWithExpression: @"vector" dimensions: 300 centroids: 10];
+    
+    [collection createIndexWithName: @"words_index" config: config error: &error];
+    
+    NSArray* names = [collection indexes: &error];
+    AssertEqual(names.count, 1u);
+    AssertEqualObjects(names, (@[@"words_index"]));
+    
+    NSString* sql = @"select meta().id, word from _default.words where vector_match(words_index, $vector, 300)";
+    CBLMutableArray* dinnerArray = [[CBLMutableArray alloc] initWithData: kDinnerVector];
+    
+    CBLQueryParameters* parameters = [[CBLQueryParameters alloc] init];
+    [parameters setArray: dinnerArray forName:@"vector"];
+    
+    CBLQuery* q = [_db createQuery: sql error: &error];
+    [q setParameters: parameters];
+    NSString* explain = [q explain: &error];
```

Comment:
More readable when having an empty line before this line as explain is not related to set parameter at all.